### PR TITLE
Cut over SearchResponse and SearchTemplateResponse to Writeable

### PR DESCRIPTION
**Pull request by javanna**
Relates to #34389

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/search/SearchResponse.java [hunk 1]
```diff
@@ -85,6 +85,26 @@
     public SearchResponse() {
     }
 
+    public SearchResponse(StreamInput in) throws IOException {
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/action/search/SearchResponse.java]: Can we also remove an empty constructor and make all class variables final here?

File: modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/SearchTemplateResponse.java [hunk 2]
```diff
@@ -48,6 +48,12 @@
     SearchTemplateResponse() {
     }
 
+    SearchTemplateResponse(StreamInput in) throws IOException {
```
Comment (mayya-sharipova) [inline modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/SearchTemplateResponse.java]: I wonder if we can remove an empty constructor and make all class variables final here? We may need to change a little `fromXContent` and have another  constructor.

**Review by mayya-sharipova — APPROVED** [review]
Thanks @javanna!  The PR looks good to me after we make changes `SearchTemplateResponse.java` and `SearchResponse.java` as outlined in the comments.

File: server/src/main/java/org/elasticsearch/action/search/SearchResponse.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/action/search/SearchResponse.java]: yes this is not so hard so I am making this change.

File: modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/SearchTemplateResponse.java (hunk 2 above)
Comment (javanna) [reply modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/SearchTemplateResponse.java]: this is possible yet it requires some work on how SearchTemplateResponse is used, as source and response are generally set after creation. I would prefer to keep this out of this PR.

**Comment by javanna** [comment]
run elasticsearch-ci/bwc

**Comment by javanna** [comment]
elasticsearch-ci/docbldesx

**Comment by javanna** [comment]
run elasticsearch-ci/docbldesx

**Comment by javanna** [comment]
run elasticsearch-ci/1

**Pull request merged**